File: WorkIt/app/src/main/java/prot3ct/workit/views/assigned_tasks/AssignedTasksPresenter.java
package prot3ct.workit.views.assigned_tasks;

import android.content.Context;

import java.util.List;

import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;
import prot3ct.workit.data.remote.TaskData;
import prot3ct.workit.data.remote.AuthData;
import prot3ct.workit.view_models.AssignedTasksListViewModel;
import prot3ct.workit.views.assigned_tasks.base.AssignedTasksContract;

public class AssignedTasksPresenter implements AssignedTasksContract.Presenter {
    private AssignedTasksContract.View view;
    private AuthData authData;
    private TaskData taskData;

    public AssignedTasksPresenter(AssignedTasksContract.View view, Context context) {
        this.view = view;
        this.authData = new AuthData(context);
        this.taskData = new TaskData(context);
    }

    @Override
    public void getAssignedTasks() {
        taskData.getAssignedTasks()
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(
                new Observer<List<AssignedTasksListViewModel>>() {
                    @Override
                    public void onSubscribe(Disposable d) {
//                        view.showDialogForLoading();
                    }

                    @Override
                    public void onNext(List<AssignedTasksListViewModel> tasks) {
                        view.setupTasksAdapter(tasks);
//                                view.notifySuccessful();
//                                view.showListJobsActivity();
                    }

                    @Override
                    public void onError(Throwable e) {
                        view.notifyError("Error ocurred when loading tasks.");
                        e.printStackTrace();
                    }

                    @Override
                    public void onComplete() {
                    }
                });
    }

    @Override
    public void removeAssignedUser(int taskId) {
        taskData.removeAssignedUser(taskId)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                        new Observer<Boolean>() {
                            @Override
                            public void onSubscribe(Disposable d) {
                            }

                            @Override
                            public void onNext(Boolean taskRequests) {
                                view.notifySuccessful("Removed task successfully");
                            }

                            @Override
                            public void onError(Throwable e) {
                                view.notifyError("Error submiting request.");
                                e.printStackTrace();
                            }

                            @Override
                            public void onComplete() {
                            }
                        });
    }

}

File: WorkIt/app/src/main/java/prot3ct/workit/views/assigned_tasks/AssignedTasksPresenter.kt
package prot3ct.workit.views.assigned_tasks

import android.content.Context
import io.reactivex.Observer
import prot3ct.workit.data.remote.AuthData
import prot3ct.workit.data.remote.TaskData
import io.reactivex.schedulers.Schedulers
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import prot3ct.workit.view_models.AssignedTasksListViewModel
import prot3ct.workit.views.assigned_tasks.base.AssignedTasksContract

class AssignedTasksPresenter(private val view: AssignedTasksContract.View, context: Context?) :
    AssignedTasksContract.Presenter {
    private val authData: AuthData
    private val taskData: TaskData
    override fun getAssignedTasks() {
        taskData.assignedTasks
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(
                object : Observer<List<AssignedTasksListViewModel>> {
                    override fun onSubscribe(d: Disposable) {
                    }

                    override fun onNext(tasks: List<AssignedTasksListViewModel>) {
                        view.setupTasksAdapter(tasks)
                    }

                    override fun onError(e: Throwable) {
                        view.notifyError("Error occurred when loading tasks.")
                        e.printStackTrace()
                    }

                    override fun onComplete() {}
                })
    }

    override fun removeAssignedUser(taskId: Int) {
        taskData.removeAssignedUser(taskId)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(
                object : Observer<Boolean?> {
                    override fun onSubscribe(d: Disposable) {}
                    override fun onNext(taskRequests: Boolean?) {
                        view.notifySuccessful("Removed task successfully")
                    }

                    override fun onError(e: Throwable) {
                        view.notifyError("Error submiting request.")
                        e.printStackTrace()
                    }

                    override fun onComplete() {}
                })
    }

    init {
        authData = AuthData(context!!)
        taskData = TaskData(context)
    }
}